Exercises/02/Begin/CMakeLists.txt

```diff
@@ -1,8 +1,16 @@
 cmake_minimum_required(VERSION 3.16)
 project(KokkosTutorial02)
-include(../../common.cmake)
 
-add_executable(02_Exercise exercise_2_begin.cpp)
-target_link_libraries(02_Exercise Kokkos::kokkos)
+# Add a custom module path for find_package
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../.cmake")
+
+find_package(Kokkos REQUIRED)
 
+if (Kokkos_ENABLE_CUDA OR Kokkos_ENABLE_HIP OR Kokkos_ENABLE_SYCL OR Kokkos_ENABLE_OPENMPTARGET OR Kokkos_ENABLE_HPX)
+    message(WARNING "${CMAKE_CURRENT_SOURCE_DIR}"
+            "A Kokkos accelerator backend is enabled, it might cause issue with the current program"
+            "Please recompile with only a host backend enabled (e.g. -DKokkos_ENABLE_OPENMP=ON)")
+endif ()
```

Comment:
can this be moved into a separate function instead of repeating it?

Reply:
The idea is that each exercise can be used as a standalone. I think factorizing with a function or macro makes the example less clear, and I wish I could avoid including a separate (but shared) file at the terminal CMakeLists.txt.

Reply:
It some way it is akin to the weird find kokkos module isn't it? I'd slightly prefer reusing code as Jakob suggested

Reply:
I had trouble finding the proper solution as the `Kokkos_DEVICES` variable is not apparent. If I do a function, I will hide the details that might be useful to copy and paste for some users. That's the reason why it is better to be transparent there.
There are only two such exercises; we will probably not have to touch the CMakeLists.txt too often, so the function is not worth it.

However, I can add a comment explaining what this conditional does for better logic readability (which is the function's main advantage for me).